src/main/java/com/ericsson/bss/cassandra/ecaudit/filter/role/RoleAuditFilter.java

```diff
@@ -71,40 +77,39 @@ public boolean isFiltered(AuditEntry logEntry)
         return isFiltered(Roles.getRoles(primaryRole), logEntry.getPermissions(), logEntry.getResource());
     }
 
+    @VisibleForTesting
     boolean isFiltered(Set<RoleResource> roles, Set<Permission> operations, IResource resource)
     {
-        for (RoleResource role : roles)
+        List<IResource> operationResourceHierarchy = getResourceHierarchy(resource);
+        for (Permission operation : operations)
         {
-            Map<String, Set<IResource>> roleOptions = whitelistCache.getWhitelist(role);
-            if (isResourceOperationWhitelisted(roleOptions, resource))
+            if(!isOperationWhitelistedOnResourceByRoles(operation, operationResourceHierarchy, roles))
             {
-                return true;
+                return false;
             }
         }
 
-        return false;
+        return true;
     }
 
-    /**
-     * Returns true if the supplied option map is white-listing the specified resource.
-     *
-     * @param roleOptions
-     *            the role options as stored in Cassandra
-     * @param operationResource
-     *            the resource being accessed
-     * @return true if the resource is white-listed, false otherwise
-     */
-    boolean isResourceOperationWhitelisted(Map<String, Set<IResource>> roleOptions, IResource operationResource)
+    private List<IResource> getResourceHierarchy(IResource primaryResource)
     {
-        Set<IResource> whitelistResources = roleOptions.get(AuditWhitelistManager.OPERATION_ALL);
-        if (whitelistResources == null)
+        List<IResource> resourceList = new ArrayList<>(3);
```

Comment:
How come "3"? In this context seems arbitrary.

Reply:
That is the typical max-depth of a resource hierarchy. like data/ks/table or functions/ks/func. I'll create a constant to make this more clear.